kernel/src/main/java/org/kframework/compile/ResolveStrict.java

```diff
@@ -193,6 +159,55 @@ public K apply(KVariable var) {
                 sentences.add(ctx);
             }
         }
+    }
+
+    public Set<Sentence> resolve(Production production, boolean sequential) {
+        long arity = production.nonterminals().size();
+        List<Integer> strictnessPositions = new ArrayList<>();
+        List<Integer> allPositions = new ArrayList<>();
+        Set<ContextAlias> aliases = new HashSet<>();
+        String attribute;
+        Set<Sentence> sentences = new HashSet<>();
+        if (sequential) {
+            attribute = production.att().get(Att.SEQSTRICT());
+        } else {
+            attribute = production.att().get(Att.STRICT());
+        }
+        if (attribute.isEmpty()) {
+            for (int i = 1; i <= arity; i++) {
+                strictnessPositions.add(i);
+            }
+            aliases.add(DEFAULT_ALIAS);
+            resolve(sequential, sentences, arity, strictnessPositions, allPositions, aliases, production);
+            allPositions.addAll(strictnessPositions);
+        } else {
+            String[] components = attribute.split(";");
+            if (components.length == 1) {
+                if (Character.isDigit(components[0].trim().charAt(0))) {
+                    aliases.add(DEFAULT_ALIAS);
+                    setPositions(components[0].trim(), strictnessPositions, arity, production);
+                } else {
+                    for (int i = 1; i <= arity; i++) {
+                        strictnessPositions.add(i);
+                    }
+                    setAliases(components[0].trim(), aliases, production);
+                }
+                resolve(sequential, sentences, arity, strictnessPositions, allPositions, aliases, production);
+                allPositions.addAll(strictnessPositions);
+            } else if (components.length % 2 == 0) {
+                for (int i = 0; i < components.length; i+=2) {
+                    setAliases(components[i].trim(), aliases, production);
+                    setPositions(components[i+1].trim(), strictnessPositions, arity, production);
+                    resolve(sequential, sentences, arity, strictnessPositions, allPositions, aliases, production);
+                    aliases.clear();
+                    allPositions.addAll(strictnessPositions);
+                    strictnessPositions.clear();
+                }
+            } else {
+                throw KEMException.compilerError("Invalid strict attribute containing multiple semicolons.", production);
```

Comment:
This error message doesn't really reflect the code, let alone the user input.
See my other comment.